lib/material/material_gui.flow

```diff
@@ -1468,4 +1478,142 @@ MOverlayT(margin : Transform<double>, radius : Transform<double>, style : Transf
 		\sz -> MGetFocusGroup(\parent -> TFrameT(margin, radius, fselectLift(style, \s -> mgraphicsStyle2tgraphicsStyle(parent, s)), sz)),
 		true
 	)
+}
+
+MSegmentedButtons(
+	items : [SegmentButtonItem],
+	style : [SegmentButtonsStyle],
+	graphicsStyles : [MGraphicsStyle]
+) -> Material {
+	MGetFocusGroup(\fg -> {
+		multiSelect = extractStruct(style, MMultiSelect(false)).enabled;
+		iconsOnly = extractStruct(style, MIconsOnly(false)).enabled;
+		defaultBgColor = extractStruct(graphicsStyles, MThemedFill(MSecondaryColor())).color;
+		defaultStrokeColor = extractStruct(graphicsStyles, MThemedStroke(either(fg.theme.focusOutlineColor, MAccentColor()))).color;
+		
+		paddingWithoutIcon = 12.0;
+
+		maxHeight = make(0.0);
+		lastSelected = make(None());
+
+		makeButton = \i, item -> {
+			buttonState = item.state;
+			isSelected = item.selected;
+			maxWidth = make(60.0);
+			
+			title = if (iconsOnly) "" else eitherFn(item.text, \text -> text.text, \ -> "");
+			textStyle = eitherFn(item.text, \text -> text.style, \ -> []);
+			iconName = eitherFn(item.icon, \icon -> icon.name, \ -> "");
+			iconStyle = eitherFn(item.icon, \icon -> icon.style, \ -> []);
+			accessibilityLabel = if (title != "") title else iconName;
+			buttonRole = if (multiSelect) "checkbox" else "radio";
+
+			isFirst = i == 0;
+			isLast = i == length(items) - 1;
+			createIcon = \icon -> MIcon(icon, concat([defaultBgColor], iconStyle)); 
+			createCheckIcon = MIcon("check", concat(iconStyle, [MIconSize(20.0), defaultBgColor])) |> MBorderRight(4.);
+
+			focused = make(false);
+
+			interactive = MComponent(
+				concat(
+					[
+						MOnClick(\ -> {
+							if (!multiSelect) {
```

Comment:
It kinda works and looks ok, but is a little complicated, as for me. I'd better just check for current value of 'item' - if true, then reverse: if false - iter through all items, set false for each, and set true for 'item'.